Compilation error with `tui` feature enabled.

If I run `cargo clippy --all-targets --all-features` getting this error:

```
error[E0599]: no method named `join` found for struct `indicatif::MultiProgress` in the current scope
   --> src/progress.rs:107:18
    |
106 | /             self.progress_group
107 | |                 .join()
    | |                 -^^^^ method not found in `MultiProgress`
    | |_________________|
    |
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `progress::Factory` defines an item `join`, perhaps you need to implement it
```

I dig a little bit and it looks like this method has been removed https://github.com/console-rs/indicatif/pull/284

I am happy to help with this, but it looks like this change will require significant change in how ProgressBar is built for Tui

--- a/examples/download.rs
+++ b/examples/download.rs
@@ -75,7 +75,7 @@ fn main() {
         .build()
         .unwrap();
 
-    let dl = downloader::Download::new("https://speed.hetzner.de/100MB.bin");
+    let dl = downloader::Download::new("https://ash-speed.hetzner.com/100MB.bin");
 
     #[cfg(not(feature = "tui"))]
     let dl = dl.progress(SimpleReporter::create());

--- a/examples/tui_basic.rs
+++ b/examples/tui_basic.rs
@@ -25,7 +25,7 @@ fn main() {
 
     // Download with an inferred filename
     let dl2 = downloader::Download::new(
-        "https://cdimage.debian.org/debian-cd/current/i386/iso-cd/debian-10.9.0-i386-netinst.iso",
+        "https://cdimage.debian.org/debian-cd/12.8.0/i386/iso-cd/debian-12.8.0-i386-netinst.iso",
     );
 
     let result = downloader.download(&[dl, dl2]).unwrap();

--- a/src/backend.rs
+++ b/src/backend.rs
@@ -163,7 +163,6 @@ pub(crate) fn run(
     downloads: Vec<Download>,
     retries: u16,
     parallel_requests: u16,
-    spin: &dyn Fn(),
 ) -> Vec<Result<DownloadSummary>> {
     let rt = tokio::runtime::Runtime::new().unwrap();
     let cl = client.clone();
@@ -176,8 +175,6 @@ pub(crate) fn run(
             .await
     });
 
-    spin();
-
     rt.block_on(result).unwrap()
 }
 
@@ -186,7 +183,6 @@ pub(crate) async fn async_run(
     downloads: Vec<Download>,
     retries: u16,
     parallel_requests: u16,
-    spin: &dyn Fn(),
 ) -> Vec<Result<DownloadSummary>> {
     let cl = client.clone();
 
@@ -198,8 +194,6 @@ pub(crate) async fn async_run(
             .await
     })
     .await;
-
-    spin();
 
     result.unwrap()
 }

--- a/src/downloader.rs
+++ b/src/downloader.rs
@@ -115,9 +115,6 @@ impl Downloader {
             to_process,
             self.retries,
             self.parallel_requests,
-            &move || {
-                factory.join();
-            },
         ))
     }
 
@@ -144,9 +141,6 @@ impl Downloader {
             to_process,
             self.retries,
             self.parallel_requests,
-            &move || {
-                factory.join();
-            },
         )
         .await;
 

--- a/src/progress.rs
+++ b/src/progress.rs
@@ -31,9 +31,6 @@ pub trait Reporter: Send + Sync {
 pub trait Factory {
     /// Create an `Reporter`
     fn create_reporter(&self) -> crate::Progress;
-
-    /// Wait for all progresses to finish
-    fn join(&self);
 }
 
 // ----------------------------------------------------------------------
@@ -66,8 +63,6 @@ impl Factory for Noop {
     fn create_reporter(&self) -> crate::Progress {
         Self::create()
     }
-
-    fn join(&self) {}
 }
 
 // ----------------------------------------------------------------------
@@ -100,13 +95,6 @@ mod tui {
                 ),
             })
         }
-
-        /// Wait for all progresses to finish
-        fn join(&self) {
-            self.progress_group
-                .join()
-                .expect("No ui if this fails, which is OK for us");
-        }
     }
 
     struct TuiBar {
@@ -120,8 +108,9 @@ mod tui {
                 lock.set_length(t);
                 lock.set_style(indicatif::ProgressStyle::default_bar()
                 .template("[{bar:20.cyan/blue}] {bytes}/{total_bytes} ({bytes_per_sec}, {eta}) - {msg}")
+                .unwrap()
                 .progress_chars("#- "));
-                lock.set_message(message);
+                lock.set_message(String::from(message));
                 lock.reset_eta();
             } else {
                 lock.set_style(
@@ -137,9 +126,10 @@ mod tui {
                             "▹▹▹▹▸",
                             "▪▪▪▪▪",
                         ])
-                        .template("{spinner:.blue} {msg}"),
+                        .template("{spinner:.blue} {msg}")
+                        .unwrap()
                 );
-                lock.set_message(message)
+                lock.set_message(String::from(message))
             };
         }
 
@@ -150,7 +140,7 @@ mod tui {
 
         fn set_message(&self, message: &str) {
             let lock = self.progress_bar.lock().unwrap();
-            lock.set_message(message);
+            lock.set_message(String::from(message));
         }
 
         fn done(&self) {

